internal/remediation/suggester/maven.go

```diff
@@ -2,23 +2,124 @@ package suggester
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"log"
+	"os"
+	"strings"
 
 	"deps.dev/util/resolve"
+	"deps.dev/util/resolve/dep"
 	"deps.dev/util/semver"
+	"github.com/google/osv-scanner/internal/resolution/manifest"
+	"github.com/google/osv-scanner/pkg/lockfile"
+	"golang.org/x/exp/slices"
 )
 
 type MavenSuggester struct{}
 
-// Suggest returns the latest version based on the given Maven requirement version.
+// Suggest returns the ManifestPatch to update Maven dependencies to a newer
+// version based on the options.
+// ManifestPatch also includes the property patches to update.
+func (ms *MavenSuggester) Suggest(ctx context.Context, client resolve.Client, mf manifest.Manifest, opts SuggestOptions) (manifest.ManifestPatch, error) {
+	specific, ok := mf.EcosystemSpecific.(manifest.MavenManifestSpecific)
+	if !ok {
+		return manifest.ManifestPatch{}, errors.New("invalid MavenManifestSpecific data")
+	}
+
+	var changedDeps []manifest.DependencyPatch
+	propertyPatches := manifest.MavenPropertyPatches{}
+	for _, req := range mf.Requirements {
+		if slices.Contains(opts.NoUpdates, req.Name) {
+			continue
+		}
+		if opts.IgnoreDev && lockfile.MavenEcosystem.IsDevGroup(mf.Groups[req.PackageKey]) {
+			// Skip the update if the dependency is of development group
+			// and updates on development dependencies are not desired
+			continue
+		}
+
+		latest, err := suggestVersion(ctx, client, req, slices.Contains(opts.AvoidMajor, req.Name))
+		if err != nil {
+			return manifest.ManifestPatch{}, fmt.Errorf("suggesting latest version of %s: %w", req.Version, err)
+		}
+		if latest.Version == req.Version {
+			// No need to update
+			continue
+		}
+
+		// TODO: this should go to reporter?
```

Comment:
+1. CC  @another-rex can you give some pointers on how to do this?

Reply:
Can this print be moved outside of the `Suggest` function? I don't think this should be printing as a side effect.

Reply:
yes - let me remove this print for now. We can print from the subcommand.